serializer.go

```diff
@@ -88,8 +88,9 @@ type NamedRef struct {
 }
 
 type Page struct {
-	Data  []any  `fauna:"data"`
 	After string `fauna:"after"`
+	Data  []any  `fauna:"data"`
+	Stats *Stats `fauna:"-"`
```

Comment:
oops looks like I never hit go on this review...

But I was thinking about it again. It's still troubling to me that we would add Stats onto the Page itself, when normally Stats are only on the response wrapping the data (a Page in this case). 

In the example file, `paginator.Next()` returns a Page. What if we provided a transform method on the iterator to return the whole response, rather than just a page?

For example, `paginatorWithStats := paginator.withStats()`, and then `pageWithStats, err := paginatorWithStats.Next()`

Names could be better and I'm certain I botched the syntax. But this could be a backwards compatible way to introduce a way to handle response stats in each of the drivers' pagination helpers, given that we intentionally stripped the iterator of the stats in the basic implementation.